### PR TITLE
Support 32/64 bits build

### PR DESCRIPTION
**Pull request by zchokri**
Add SUPPORT_32BIT_IOCTL to CCFLAGS to support 32 bit ioctl system call

### CODE REVIEW

**Comment by andrew-elder** [comment]
@zchokri - thank you for this pull request. I have a few questions that I will put in the "changes".

File: daemons/gptp/common/ieee1588.hpp [hunk 1]
```diff
@@ -379,9 +379,11 @@ class Timestamp {
...
 #define PDELAY_PENDING_TIMESTAMP (Timestamp( 0xC0000001, 0, 0 ))	/*!< PDelay is pending timestamp */
 
-#define TIMESTAMP_TO_NS(ts) (((static_cast<long long int>((ts).seconds_ms) \
-			       << sizeof((ts).seconds_ls)*8) + \
-			      (ts).seconds_ls)*1000000000LL + (ts).nanoseconds)		/*!< Converts timestamp value into nanoseconds value*/
+static inline uint64_t TIMESTAMP_TO_NS(Timestamp &ts)
+{
+	return (((static_cast<long long int>((ts).seconds_ms) << sizeof((ts).seconds_ls)*8) +
+			      (ts).seconds_ls)*1000000000LL + (ts).nanoseconds)	;	/*!< Converts timestamp value into nanoseconds value*/
+}
```
Comment (andrew-elder) [inline daemons/gptp/common/ieee1588.hpp]: Was this change required for building the 32-bit version?

File: daemons/gptp/common/ptptypes.hpp [hunk 2]
```diff
@@ -36,7 +36,12 @@
 
 /**@file*/
 
+#if defined(__clang__) &&  defined(__x86_64__)
+// Clang/llvm has incompatible long double (fp128) for x86_64.
+typedef double FrequencyRatio; /*!< Frequency Ratio */
+#else
```
Comment (andrew-elder) [inline daemons/gptp/common/ptptypes.hpp]: This should be a separate commit? It seems un-related to getting 32-bit to compile and run.

File: daemons/gptp/linux/shm_test/shm_test.cpp [hunk 3]
```diff
@@ -97,11 +97,12 @@ int main(int argc, char *argv[])
     fprintf(stdout, "ml freq offset %Lf\n", ptpData->ml_freqoffset);
     fprintf(stdout, "ls phoffset %ld\n", ptpData->ls_phoffset);
     fprintf(stdout, "ls freq offset %Lf\n", ptpData->ls_freqoffset);
-    fprintf(stdout, "local time %lu\n", ptpData->local_time);
+    fprintf(stdout, "local time %llu\n", ptpData->local_time);
     fprintf(stdout, "sync count %u\n", ptpData->sync_count);
     fprintf(stdout, "pdelay count %u\n", ptpData->pdelay_count);
     fprintf(stdout, "asCapable %s\n", ptpData->asCapable ? "True" : "False");
     fprintf(stdout, "Port State %d\n", (int)ptpData->port_state);
+    fprintf(stdout, "process_id %d\n", (int)ptpData->process_id);
```
Comment (andrew-elder) [inline daemons/gptp/linux/shm_test/shm_test.cpp]: I think commits in this file should also be a separate commit - I do support these changes.

File: lib/igb/igb.c [hunk 4]
```diff
@@ -1837,7 +1837,10 @@ struct timespec timespec_addns(struct timespec *a, unsigned long addns)
 	return *a;
 }
 
-#define TS2NS(ts) (((ts).tv_sec*1000000000)+(ts).tv_nsec)
+static inline u_int32_t TS2NS(struct timespec ts)
+{
+    return (((ts).tv_sec*1000000000)+(ts).tv_nsec);
+}
```
Comment (andrew-elder) [inline lib/igb/igb.c]: Again - was this required for the 32-bit build? Perhaps a separate commit would be better. Also, the (ts).tv_sec could be replaced with ts.tv_sec if you are now using an inline function.

File: kmod/igb/Makefile [hunk 5]
```diff
@@ -41,6 +41,11 @@ endif
 # PTP is required for AVB support
 CFLAGS_EXTRA += -DIGB_PTP
 
+# Add 32 bit ioctl support
+ifeq ($(32BIT_IOCTL_SUPPORT),y)
+CFLAGS_EXTRA += -DSUPPORT_32BIT_IOCTL
+endif
+
```
Comment (andrew-elder) [inline kmod/igb/Makefile]: This change seems to me to be the main one relevant to the title of this pull request. I think it should be a separate commit.

File: daemons/gptp/common/ptptypes.hpp (hunk 2 above)
Comment (zchokri) [inline daemons/gptp/common/ptptypes.hpp]: done in 3e91a630455c509310bed2e8d93226347025a849

**Comment by andrew-elder** [comment]
Reviewed on conf call.
